src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.JS.Owned.cs

```diff
@@ -10,7 +10,7 @@ public static partial class Runtime
     {
         private static object JSOwnedObjectLock = new object();
         // we use this to maintain identity of GCHandle for a managed object
-        private static Dictionary<object, int> GCHandleFromJSOwnedObject = new Dictionary<object, int>();
+        private static Dictionary<object, IntPtr> GCHandleFromJSOwnedObject = new Dictionary<object, IntPtr>();
```

Comment:
I just noticed that we're not doing a reference compare here. Please add a todo comment or fix it (I think the default behavior here will depend on .Equals of the object, but we want object.ReferenceEquals in all cases.)

Reply:
great catch!